dask_planner/src/sql.rs

```diff
@@ -576,3 +571,55 @@ impl PlanVisitor for OptimizablePlanVisitor {
         Ok(true)
     }
 }
+
+fn generate_numeric_signatures(n: i32) -> Signature {
```

Comment:
This works well for uniform signatures that accept int/float inputs; however, as we add in timestamp functions, it would be nice to have a utility like this be flexible for an arbitrary list of valid input types.

I was thinking maybe a function that accepts a vector of vectors of `DataTypes` and returns the corresponding cartesian product, i.e. something like:

```rust
let sig = generate_signatures(vec![
    vec![DataType::Int64, DataType::Float64],
    vec![DataType::Utf8, DataType::Int64]
]);
// would be equivalent to
// let sig = Signature::one_of(
//     vec![
//         TypeSignature::Exact(vec![DataType::Int64, DataType::Utf8]),
//         TypeSignature::Exact(vec![DataType::Int64, DataType::Int64]),
//         TypeSignature::Exact(vec![DataType::Float64, DataType::Utf8]),
//         TypeSignature::Exact(vec![DataType::Float64, DataType::Int64]),
//     ],
//     Volatility::Immutable,
// );
```